.github/scripts/generate_pytorch_test_matrix.py

```diff
@@ -22,7 +22,11 @@ class Config(TypedDict):
 def main() -> None:
     TEST_RUNNER_TYPE = os.getenv('TEST_RUNNER_TYPE')
     assert TEST_RUNNER_TYPE is not None
+    ON_PULL_REQUEST = os.getenv('GITHUB_HEAD_REF')
+    NUM_TEST_SHARDS_ON_PULL_REQUEST = os.getenv('NUM_TEST_SHARDS_ON_PULL_REQUEST')
     NUM_TEST_SHARDS = int(os.getenv('NUM_TEST_SHARDS', '1'))
+    if ON_PULL_REQUEST and NUM_TEST_SHARDS_ON_PULL_REQUEST:
```

Comment:
how about we just use SMOKE_TEST_ONLY keyword?
```
if ON_PULL_REQUEST and SMOKE_TEST_ONLY:
    NUM_TEST_SHARDS = 1
```

Reply:
I considered this, but I wanted as much decoupling as possible, and it is more clear with NUM_TEST_SHARDS_ON_PULL_REQUEST than SMOKE_TEST_ONLY logic across several places in the codebase.